joint_trajectory_controller/src/joint_trajectory_controller.cpp

```diff
@@ -84,32 +85,31 @@ JointTrajectoryController::update()
     return CONTROLLER_INTERFACE_RET_SUCCESS;
   }
 
-  // when no traj msg has been received yet
-  if (!traj_point_active_ptr_ || (*traj_point_active_ptr_)->is_empty()) {
-    return CONTROLLER_INTERFACE_RET_SUCCESS;
-  }
+  // when we receive a traj msg
+  if (traj_point_active_ptr_ && (*traj_point_active_ptr_)->is_empty() == false) {
+    // find next new point for current timestamp
+    auto traj_point_ptr = (*traj_point_active_ptr_)->sample(lifecycle_node_->now());
+    // find next new point for current timestamp
+    // set cmd only if a point is found
+    if (traj_point_ptr == (*traj_point_active_ptr_)->end()) {
+      return CONTROLLER_INTERFACE_RET_SUCCESS;
+    }
 
-  // find next new point for current timestamp
-  auto traj_point_ptr = (*traj_point_active_ptr_)->sample(rclcpp::Clock().now());
-  // find next new point for current timestamp
-  // set cmd only if a point is found
-  if (traj_point_ptr == (*traj_point_active_ptr_)->end()) {
-    return CONTROLLER_INTERFACE_RET_SUCCESS;
-  }
+    // check if new point ptr points to the same as previous point
+    if (prev_traj_point_ptr_ == traj_point_ptr) {
+      return CONTROLLER_INTERFACE_RET_SUCCESS;
+    }
 
-  // check if new point ptr points to the same as previous point
-  if (prev_traj_point_ptr_ == traj_point_ptr) {
-    return CONTROLLER_INTERFACE_RET_SUCCESS;
-  }
+    size_t joint_num = registered_joint_cmd_handles_.size();
+    for (size_t index = 0; index < joint_num; ++index) {
+      registered_joint_cmd_handles_[index]->set_cmd(traj_point_ptr->positions[index]);
+    }
 
-  size_t joint_num = registered_joint_cmd_handles_.size();
-  for (size_t index = 0; index < joint_num; ++index) {
-    registered_joint_cmd_handles_[index]->set_cmd(traj_point_ptr->positions[index]);
+    prev_traj_point_ptr_ = traj_point_ptr;
+    set_op_mode(hardware_interface::OperationMode::ACTIVE);
```

Comment:
In fact, I generally disagree with this _change_. It's only really indentation and whitespaces. The only diff here should be the call to `publish_state()`.

Reply:
see comment above